ipv8/REST/health_endpoint.py

```diff
@@ -0,0 +1,107 @@
+from ..peerdiscovery.discovery import DiscoveryStrategy
+
+
+class DriftMeasurementStrategy(DiscoveryStrategy):
+
+    def __init__(self, core_update_rate):
+        super(DriftMeasurementStrategy, self).__init__(None)
+        self.last_measurement = time.time()
+        self.history = collections.deque(maxlen=100)
+        self.core_update_rate = core_update_rate
+        self.overlay = type('FakeOverlay', (object,), {'get_peers': lambda: []})
+
+    def take_step(self):
+        with self.walk_lock:
+            this_time = time.time()
+            self.history.append((this_time, max(0.0, this_time - self.last_measurement - self.core_update_rate)))
+            self.last_measurement = this_time
+
+
+class HealthEndpoint(BaseEndpoint):
+    """
+    This endpoint manages measurements of non-functional requirements.
+    """
+
+    def __init__(self):
+        super(HealthEndpoint, self).__init__()
+        self.strategy = None
+        self.enabled = False
+
+    def setup_routes(self):
+        self.app.add_routes([web.get('/drift', self.retrieve_drift), web.put('/enable', self.enable_measurements)])
+
+    def enable(self):
+        if not self.session:
+            return False
+        if not self.enabled:
+            self.strategy = DriftMeasurementStrategy(self.session.walk_interval)
+            with self.session.overlay_lock:
+                self.session.strategies.append((self.strategy, -1))
+            return True
+        return True
+
+    def disable(self):
+        if not self.session:
+            return False
+        if self.enabled:
+            with self.session.overlay_lock:
+                self.session.strategies = [s for s in self.session.strategies
+                                           if not isinstance(s[0], DriftMeasurementStrategy)]
+            return True
+        return True
+
+    @docs(
+        tags=["Health"],
+        summary="Measure the core drift.",
+        responses={
+            200: {
+                "schema": schema(DriftResponse={"measurements": [
+                    schema(Measurement={
+                        "timestamp": Float,
+                        "drift": Float
+                    })
+                ]})
+            },
+            400: {
+                "schema": DefaultResponseSchema,
+                "example": {"success": False, "error": "Core drift disabled."}
+            }
+        }
+    )
+    async def retrieve_drift(self, _):
+        if self.strategy is None:
+            return Response({"success": False, "error": "Core drift disabled."}, status=HTTP_NOT_FOUND)
+        with self.strategy.walk_lock:
+            return Response({"measurements": [{"timestamp": e[0], "drift": e[1]} for e in self.strategy.history]})
+
+    @docs(
+        tags=["Health"],
+        summary="Enable or disable measurements.",
+        responses={
+            200: {
+                "schema": DefaultResponseSchema,
+                "examples": {"success": True}
```

Comment:
You need to change `examples` to `example`, otherwise 'success' will be considered the name of the example (instead of a JSON field).